### ISSUE TITLE
Analysis import fails due to celery SoftTimeLimitExceeded 

### ISSUE THREAD

**Issue by gmnelson** [refinery-platform/refinery-platform#3435]
* Specific code commit: 3abb3aa
* Version of the web browser and OS: chrome, macOS
* Environment where the error occurred (Vagrant VM and site conf mode or AWS instance): hsci.stemcellcommons.org 

### Steps to reproduce
Please list all the actions and the input data used:

1. Launch analysis that completes successfully in Galaxy

### Observed behavior
* Analysis still fails Refinery-side, **without email notification**
* Logs suggest celery SoftTimeLimitExceeded while importing outputs back into Refinery

### Expected behavior
* Outputs should import successfully and analysis should register as successful in Refinery

**Comment by jkmarx** [comment]
Traceback from celery:
```2019-08-29 21:15:20,461 ERROR    analysis_manager.tasks:51 on_failure[....4856fa] - Setting status of analysis 'ChIP-seq Peak Calling - Human - 2019/08/29 15:50:17 - Gnelson - gnelson - Galaxy workflow execution for: ChIP-seq Peak Calling - Human' to failure
2019-08-29 21:15:20,484 ERROR    celery.worker.job:282 log() - Task analysis_manager.tasks.run_analysis[.....4856fa] raised unexpected: SoftTimeLimitExceeded()
Traceback (most recent call last):
  File "/home/ubuntu/.virtualenvs/refinery-platform/local/lib/python2.7/site-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/home/ubuntu/.virtualenvs/refinery-platform/local/lib/python2.7/site-packages/celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
  File "/srv/refinery-platform/refinery/analysis_manager/tasks.py", line 303, in run_analysis
    _galaxy_file_export(analysis_uuid)
  File "/srv/refinery-platform/refinery/analysis_manager/tasks.py", line 184, in _galaxy_file_export
    galaxy_export_tasks = _get_galaxy_download_task_ids(analysis)
  File "/srv/refinery-platform/refinery/analysis_manager/tasks.py", line 462, in _get_galaxy_download_task_ids
    tool.create_analysis_output_node_connections()
  File "/srv/refinery-platform/refinery/tool_manager/models.py", line 818, in create_analysis_output_node_connections
    for galaxy_dataset in self._get_galaxy_history_dataset_list():
  File "/srv/refinery-platform/refinery/tool_manager/models.py", line 652, in func_wrapper
    return func(*args, **kwargs)
  File "/srv/refinery-platform/refinery/tool_manager/models.py", line 1103, in _get_galaxy_history_dataset_list
    self._get_workflow_step(galaxy_dataset) > 0
  File "/srv/refinery-platform/refinery/tool_manager/models.py", line 1268, in _get_workflow_step
    for step in self._get_galaxy_workflow_invocation()["steps"]:
  File "/srv/refinery-platform/refinery/tool_manager/models.py", line 652, in func_wrapper
    return func(*args, **kwargs)
  File "/srv/refinery-platform/refinery/tool_manager/models.py", line 1178, in _get_galaxy_workflow_invocation
    self.get_galaxy_dict()[self.GALAXY_WORKFLOW_INVOCATION_DATA]["id"]
  File "/home/ubuntu/.virtualenvs/refinery-platform/local/lib/python2.7/site-packages/bioblend/galaxy/workflows/__init__.py", line 515, in show_invocation
    return self._get(url=url)
  File "/home/ubuntu/.virtualenvs/refinery-platform/local/lib/python2.7/site-packages/bioblend/galaxy/client.py", line 115, in _get
    r = self.gi.make_get_request(url, params=params)
  File "/home/ubuntu/.virtualenvs/refinery-platform/local/lib/python2.7/site-packages/bioblend/galaxyclient.py", line 91, in make_get_request
    r = requests.get(url, **kwargs)
  File "/home/ubuntu/.virtualenvs/refinery-platform/local/lib/python2.7/site-packages/requests/api.py", line 75, in get
    return request('get', url, params=params, **kwargs)
  File "/home/ubuntu/.virtualenvs/refinery-platform/local/lib/python2.7/site-packages/requests/api.py", line 60, in request
    return session.request(method=method, url=url, **kwargs)
  File "/home/ubuntu/.virtualenvs/refinery-platform/local/lib/python2.7/site-packages/requests/sessions.py", line 533, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/ubuntu/.virtualenvs/refinery-platform/local/lib/python2.7/site-packages/requests/sessions.py", line 646, in send
    r = adapter.send(request, **kwargs)
  File "/home/ubuntu/.virtualenvs/refinery-platform/local/lib/python2.7/site-packages/requests/adapters.py", line 449, in send
    timeout=timeout
  File "/home/ubuntu/.virtualenvs/refinery-platform/local/lib/python2.7/site-packages/urllib3/connectionpool.py", line 600, in urlopen
    chunked=chunked)
  File "/home/ubuntu/.virtualenvs/refinery-platform/local/lib/python2.7/site-packages/urllib3/connectionpool.py", line 377, in _make_request
    httplib_response = conn.getresponse(buffering=True)
  File "/usr/lib/python2.7/httplib.py", line 1136, in getresponse
    response.begin()
  File "/usr/lib/python2.7/httplib.py", line 453, in begin
    version, status, reason = self._read_status()
  File "/usr/lib/python2.7/httplib.py", line 409, in _read_status
    line = self.fp.readline(_MAXLINE + 1)
  File "/usr/lib/python2.7/socket.py", line 480, in readline
    data = self._sock.recv(self._rbufsize)
  File "/home/ubuntu/.virtualenvs/refinery-platform/local/lib/python2.7/site-packages/billiard/pool.py", line 235, in soft_timeout_sighandler
    raise SoftTimeLimitExceeded()
SoftTimeLimitExceeded: SoftTimeLimitExceeded()
```

**Comment by ilan-gold** [comment]
From looking at the stacktrace and getting a grasp on what is going on, it seems the issue is getting the tasks (and the overhead involved with that) that correspond to file downloads from galaxy and not actually importing the data itself from galaxy (which is a FileImportTask subtask and therefore not subject to the same 60 second SoftTimeLimitExceeded).  Without access to the logs, it's tough to say.  I will look at those tomorrow/try to reproduce by using a large number of small files.

**Comment by jkmarx** [comment]
Same error, different breaking point. Analysis ran successfully in galaxy.
```2019-09-13 11:31:19,043 ERROR    celery.worker.job:282 log() - Task analysis_manager.tasks.run_analysis[........2a190] raised unexpected: SoftTimeLimitExceeded()
Traceback (most recent call last):
  File "/home/ubuntu/.virtualenvs/refinery-platform/local/lib/python2.7/site-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/home/ubuntu/.virtualenvs/refinery-platform/local/lib/python2.7/site-packages/celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
  File "/srv/refinery-platform/refinery/analysis_manager/tasks.py", line 304, in run_analysis
    _attach_workflow_outputs(analysis_uuid)
  File "/srv/refinery-platform/refinery/analysis_manager/tasks.py", line 151, in _attach_workflow_outputs
    analysis.attach_derived_nodes_to_dataset()
  File "/srv/refinery-platform/refinery/core/models.py", line 1402, in attach_derived_nodes_to_dataset
    self._create_annotated_nodes()
  File "/srv/refinery-platform/refinery/core/models.py", line 1671, in _create_annotated_nodes
    self._prepare_annotated_nodes(node_uuids)
  File "/srv/refinery-platform/refinery/core/models.py", line 1459, in _prepare_annotated_nodes
    self.rename_results()
  File "/srv/refinery-platform/refinery/core/models.py", line 1379, in rename_results
    item.rename_datafile(result.file_name)
  File "/srv/refinery-platform/refinery/file_store/models.py", line 179, in rename_datafile
    storage.bucket_name, new_file_store_name)
  File "/srv/refinery-platform/refinery/file_store/utils.py", line 112, in copy_s3_object
    ExtraArgs=S3_WRITE_ARGS, Callback=progress_report)
  File "/home/ubuntu/.virtualenvs/refinery-platform/local/lib/python2.7/site-packages/boto3/s3/inject.py", line 267, in copy
    return future.result()
  File "/home/ubuntu/.virtualenvs/refinery-platform/local/lib/python2.7/site-packages/s3transfer/futures.py", line 73, in result
    return self._coordinator.result()
  File "/home/ubuntu/.virtualenvs/refinery-platform/local/lib/python2.7/site-packages/s3transfer/futures.py", line 228, in result
    self._done_event.wait(MAXINT)
  File "/usr/lib/python2.7/threading.py", line 614, in wait
    self.__cond.wait(timeout)
  File "/usr/lib/python2.7/threading.py", line 359, in wait
    _sleep(delay)
  File "/home/ubuntu/.virtualenvs/refinery-platform/local/lib/python2.7/site-packages/billiard/pool.py", line 235, in soft_timeout_sighandler
    raise SoftTimeLimitExceeded()
SoftTimeLimitExceeded: SoftTimeLimitExceeded()
```

**Comment by ilan-gold** [comment]
So how do you think we should proceed?  Handle the bug, exit gracefully, and retry?  Or we could just bump up our time limit (CELERYD_TASK_SOFT_TIME_LIMIT in [config/settings/base.py](https://github.com/refinery-platform/refinery-platform/blob/develop/refinery/config/settings/base.py)) to something like two or three minutes and hope that is enough.  I think doing both might be a good option, along with an email notification maybe to notify if the timeout is ever breached.

**Comment by ilan-gold** [comment]
There are two things that can be done here
1) Fix #2642 
2) Make requests aysnchronously once we have #2619, and we can use an executor pool https://docs.python.org/dev/library/concurrent.futures.html#concurrent.futures.ThreadPoolExecutor
